Crash in destroying Zerox

Edwood got confused when I tried to delete a Zerox. I think the issue is a similar crash to one that I've seen before: middle click doesn't always successfully capture a valid string. However, in this case, we crash in `File` because the code for removing `Text` observers checks that one is attempting to remove an observer that actually exists.

Trace below. My ongoing work for #97 will make this go away but that would conceivably only hide an underlying issue with middle click.

```
2018/12/01 19:32:26 acme: can't find text in File.DelText: <nil>
panic: acme: can't find text in File.DelText: <nil>


goroutine 10 [running, locked to thread]:
log.Panicf(0x1201866, 0xd, 0xc0005c7d80, 0x2, 0x2)
        /usr/local/Cellar/go/1.11.1/libexec/src/log/log.go:333 +0xda
main.acmeerror(0x12068ac, 0x1f, 0x0, 0x0)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/util.go:38 +0xd2
main.(*File).DelText(0xc000152d00, 0xc00007f928)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/file.go:104 +0x13b
main.(*Text).Close(0xc00007f928)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/text.go:183 +0x54
main.(*Window).Close(0xc00007f800)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/wind.go:377 +0xa1
main.(*Window).Unlock(0xc00007f800)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/wind.go:355 +0x64
main.MovedMouse(0xa1, 0x1ee, 0x2, 0x6c20a7a1)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/acme.go:443 +0x3db
main.mousethread(0xc0000ea0c0)
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/acme.go:301 +0x2c4
created by main.main
        /Users/rjkroege/tools/gopkg/src/github.com/rjkroege/edwood/acme.go:172 +0x8fc
```

--- a/file.go
+++ b/file.go
@@ -107,17 +107,18 @@ func (f *File) AddText(t *Text) *File {
 }
 
 func (f *File) DelText(t *Text) {
-
 	for i, text := range f.text {
 		if text == t {
 			f.text[i] = f.text[len(f.text)-1]
 			f.text = f.text[:len(f.text)-1]
 			if len(f.text) == 0 {
+				f.Close()
 				return
 			}
 			if t == f.curtext {
 				f.curtext = f.text[0]
 			}
+			return
 		}
 	}
 	acmeerror("can't find text in File.DelText", nil)

--- a/file_test.go
+++ b/file_test.go
@@ -4,7 +4,7 @@ import "testing"
 
 func TestDelText(t *testing.T) {
 	f := &File{
-		text: []*Text{&Text{}, &Text{}, &Text{}, &Text{}, &Text{}},
+		text: []*Text{{}, {}, {}, {}, {}},
 	}
 	t.Run("Nonexistent", func(t *testing.T) {
 		defer func() {

--- a/file_test.go
+++ b/file_test.go
@@ -1,0 +1,29 @@
+package main
+
+import "testing"
+
+func TestDelText(t *testing.T) {
+	f := &File{
+		text: []*Text{&Text{}, &Text{}, &Text{}, &Text{}, &Text{}},
+	}
+	t.Run("Nonexistent", func(t *testing.T) {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Errorf("expected panic when deleting nonexistent text")
+			}
+		}()
+		f.DelText(&Text{})
+	})
+	for i := len(f.text) - 1; i >= 0; i-- {
+		text := f.text[i]
+		f.DelText(text)
+		if got, want := len(f.text), i; got != want {
+			t.Fatalf("DelText resulted in text of length %v; expected %v", got, want)
+		}
+		for i, t1 := range f.text {
+			if t1 == text {
+				t.Fatalf("DelText did not delete correctly at index %v", i)
+			}
+		}
+	}
+}

